src/nbl/asset/interchange/CGLTFLoader.h

```diff
@@ -0,0 +1,539 @@
+								The alphaMode property specifies how alpha is interpreted. These values are linear. If a baseColorTexture is specified, this value is multiplied with the texel values.
+							*/
+
+							std::optional<std::array<double, 4>> baseColorFactor;
+
+							/*
+								The base color texture. The first three components (RGB) are encoded with the sRGB transfer function. They specify the base color of the material. If the fourth component (A) is present, it represents the linear alpha coverage of the material.
+								Otherwise, an alpha of 1.0 is assumed. The alphaMode property specifies how alpha is interpreted. The stored texels must not be premultiplied.
+							*/
+
+							std::optional<SBaseColorTexture> baseColorTexture;
+
+							/*
+								The metalness of the material. A value of 1.0 means the material is a metal. A value of 0.0 means the material is a dielectric. Values in between are for blending between metals and dielectrics such as dirty metallic surfaces.
+								This value is linear. If a metallicRoughnessTexture is specified, this value is multiplied with the metallic texel values.
+							*/
+
+							std::optional<double> metallicFactor;
+
+							/*
+								The roughness of the material. A value of 1.0 means the material is completely rough. A value of 0.0 means the material is completely smooth. 
+								This value is linear. If a metallicRoughnessTexture is specified, this value is multiplied with the roughness texel values.
+							*/
+
+							std::optional<double> roughnessFactor;
+
+							/*
+								The metallic-roughness texture. The metalness values are sampled from the B channel. The roughness values are sampled from the G channel. 
+								These values are linear. If other channels are present (R or A), they are ignored for metallic-roughness calculations.
+							*/
+
+							std::optional<SMetallicRoughnessTexture> metallicRoughnessTexture;
+						};
+
+						struct SNormalTexture : public STextureInfo
+						{
+							std::optional<double> scale; //! The scalar multiplier applied to each normal vector of the normal texture.
+						};
+
+						struct SOcclusionTexture : public STextureInfo
+						{
+							std::optional<double> strength; //! A scalar multiplier controlling the amount of occlusion applied.
+						};
+
+						struct SEmissiveTexture : public STextureInfo
+						{
+							
+						};
+
+						struct SAlphaMode
+						{
+							_NBL_STATIC_INLINE_CONSTEXPR std::string_view OPAQUE_MODE = "OPAQUE";
+							_NBL_STATIC_INLINE_CONSTEXPR std::string_view MASK_MODE = "MASK";
+							_NBL_STATIC_INLINE_CONSTEXPR std::string_view BLEND_MODE = "BLEND";
+						};
+
+						std::optional<std::string> name;
+						std::optional<SPBRMetalicRoughness> pbrMetallicRoughness;
+						std::optional<SNormalTexture> normalTexture;
+						std::optional<SOcclusionTexture> occlusionTexture;
+						std::optional<SEmissiveTexture> emissiveTexture;
+						std::optional<std::array<double, 3>> emissiveFactor;
```

Comment:
should probably go into the `SEmissiveTexture` like the occlusion and normal factors go

Reply:
you probably dont need to use doubles for this